### PR TITLE
Added some functions to ReHLDS API

### PR DESCRIPTION
**Pull request by s1lentq**
and implemented unregisterHook for hookchain

### CODE REVIEW

File: rehlds/public/rehlds/rehlds_api.h [hunk 1]
```diff
@@ -117,6 +117,26 @@ typedef IVoidHookChainRegistry<model_t*, void*> IRehldsHookRegistry_Mod_LoadBrus
 typedef IVoidHookChain<model_t*, void*> IRehldsHook_Mod_LoadStudioModel;
 typedef IVoidHookChainRegistry<model_t*, void*> IRehldsHookRegistry_Mod_LoadStudioModel;
 
+//SV_EmitEvents hook
+typedef IVoidHookChain<client_t *, packet_entities_t *, sizebuf_t *> IRehldsHook_SV_EmitEvents;
```
Comment (dreamstalker) [inline rehlds/public/rehlds/rehlds_api.h]: Please don't pass client_t to 3rd party plugins. This type is private to the engine and therefore is not defined in SDK.
Use IGameClient instead.

File: rehlds/rehlds/hookchains_impl.cpp [hunk 2]
```diff
@@ -50,4 +50,20 @@ void AbstractHookChainRegistry::addHook(void* hookFunc) {
 	}
 
 	m_Hooks[m_NumHooks++] = hookFunc;
-}
\ No newline at end of file
+}
+
+void AbstractHookChainRegistry::removeHook(void* hookFunc) {
+	if (m_NumHooks <= 0) {
```
Comment (dreamstalker) [inline rehlds/rehlds/hookchains_impl.cpp]: What is the purpose of this 'if' block? m_NumHooks <= 0 cases will not break the for loop logic.

File: rehlds/public/rehlds/rehlds_api.h (hunk 1 above)
Comment (s1lentq) [inline rehlds/public/rehlds/rehlds_api.h]: all done.

File: rehlds/engine/pr_cmds.cpp [hunk 3]
```diff
@@ -1172,8 +1177,18 @@ short unsigned int EXT_FUNC EV_Precache(int type, const char *psz)
...
+}
+
+void EV_PlayReliableEvent(client_t *cl, int entindex, short unsigned int eventindex, float delay, event_args_t *pargs)
+{
+	g_RehldsHookchains.m_EV_PlayReliableEvent.callChain(EV_PlayReliableEvent_api, GetRehldsApiClient(cl), entindex, eventindex, delay, pargs);
+}
+
 /* <796ae> ../engine/pr_cmds.c:1531 */
-void EXT_FUNC EV_PlayReliableEvent(client_t *cl, int entindex, short unsigned int eventindex, float delay, event_args_t *pargs)
+void EXT_FUNC EV_PlayReliableEvent_internal(client_t *cl, int entindex, short unsigned int eventindex, float delay, event_args_t *pargs)
```
Comment (dreamstalker) [inline rehlds/engine/pr_cmds.cpp]: This function is not callable from outside, so you can remove EXT_FUNC  from it.
It is also not necessary to declare it in a header file.

Comment (s1lentq) [inline rehlds/engine/pr_cmds.cpp]: done

**Pull request merged**